packages/psss-server/src/models/PsssSession.ts

```diff
@@ -4,19 +4,107 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import jwt from 'jsonwebtoken';
 import { DatabaseModel, DatabaseType } from '@tupaia/database';
 import { AccessPolicy } from '@tupaia/access-policy';
 
+import { AccessPolicyObject, SessionColumns, SessionDetails } from '../types';
+import { AuthConnection } from '../connections/AuthConnection';
+
+const getTokenExpiry = (accessToken: string) => {
+  const { exp: expiryAuthServerClock, iat: issuedAtAuthServerClock } = jwt.decode(accessToken);
+  // subtract 3 seconds to account for latency since generation on the auth server
+  const validForSeconds = expiryAuthServerClock - issuedAtAuthServerClock - 3;
+  const expiryPsssServerClock = Date.now() + validForSeconds * 1000;
```

Comment:
I have a question about that but I'll ask offline

Reply:
Darn I never got back to you on this...

Reply:
No worries, it seems like you know what you're doing here - it would be mainly for my understanding. Can do post merge